homeassistant/components/overkiz/coordinator.py

```diff
@@ -101,11 +102,27 @@ async def _async_update_data(self) -> dict[str, Device]:
             if event_handler := EVENT_HANDLERS.get(event.name):
                 await event_handler(self, event)
 
+        # Reset update interval to previous value
+        # when all executions are finished
         if not self.executions:
-            self.update_interval = UPDATE_INTERVAL
+            self._restore_update_interval()
 
         return self.devices
 
+    def set_update_interval(self, update_interval: timedelta) -> None:
+        """Store update interval if changed and set new update interval."""
+        if (
+            self.update_interval
+            and self.original_update_interval is not self.update_interval
+        ):
+            self.original_update_interval = self.update_interval
```

Comment:
This doesn't look correct.

Reply:
I will look into this. @allenporter also noticed this, so I will have a better look and rework this PR.

For now I will mark it as a draft.